app/controllers/checkout_controller.rb

```diff
@@ -1,13 +1,10 @@
 class CheckoutController < ApplicationController
   def index
-    if Order.last == nil
-      redirect_to root_path
-      flash[:notice] = "You have no items in your order"
-    elsif Order.last.shopping_cart_items.count == 0
+    if Order.where(bought: false).count == 0 || Order.find_by(bought: false).shopping_cart_items.count == 0
       redirect_to root_path
       flash[:notice] = "You have no items in your order"
     else
-      @order_items = Order.last.shopping_cart_items.all
+      @order_items = Order.find_by(bought: false).shopping_cart_items
```

Comment:
This is also a bit tricky. How do you know that there are no two currently open orders saved in the db? Exposing this in a test is a bit tricky but I bet you that if you have two concurrent users this logic will fail. (I might be wrong, but there's to much unknowns here...)